controllers/constraintstatus_controller.go

```diff
@@ -0,0 +1,284 @@
+	return ctrl.NewControllerManagedBy(mgr).
+		// The work queue prevents the same item being reconciled concurrently:
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/1416#issuecomment-899833144
+		WithOptions(controller.Options{MaxConcurrentReconciles: int(r.MaxConcurrentReconciles)}).
+		Owns(&v1alpha1.Config{}).
+		Named(ControllerName).
+		For(&v1beta1.ConstraintPodStatus{},
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					obj := e.Object.(*v1beta1.ConstraintPodStatus)
+
+					return slices.Contains(obj.Status.Operations, "audit")
+				},
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					oldObj := e.ObjectOld.(*v1beta1.ConstraintPodStatus)
+					newObj := e.ObjectNew.(*v1beta1.ConstraintPodStatus)
+
+					return slices.Contains(newObj.Status.Operations, "audit") &&
+						oldObj.Status.ObservedGeneration != newObj.Status.ObservedGeneration
+				},
+				DeleteFunc: func(e event.DeleteEvent) bool {
+					obj := e.Object.(*v1beta1.ConstraintPodStatus)
+
+					return slices.Contains(obj.Status.Operations, "audit")
+				},
+			},
+			)).
+		Complete(r)
+}
+
+// Reconcile implements reconcile.Reconciler.
+func (r *ConstraintStatusReconciler) Reconcile(ctx context.Context,
+	request reconcile.Request,
+) (reconcile.Result, error) {
+	log := r.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	log.Info("Reconciling ConstraintPodStatus and Config")
+
+	gatekeeper := &operatorv1alpha1.Gatekeeper{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: "",
+		Name:      "gatekeeper",
+	}, gatekeeper)
+
+	auditFromCache := *gatekeeper.Spec.Audit.AuditFromCache
+
+	if err != nil || !strings.EqualFold(string(auditFromCache), "Automatic") {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("auditFromCache is not set so skip this reconcile")
+	}
+
+	// Get config or create if not exist
+	config := &v1alpha1.Config{}
+	err = r.Get(ctx, types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      "config",
+	}, config)
+
+	if apierrors.IsNotFound(err) {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: r.Namespace,
+			},
+			Spec: v1alpha1.ConfigSpec{
+				Validation: v1alpha1.Validation{},
+				Sync: v1alpha1.Sync{
+					SyncOnly: []v1alpha1.SyncOnlyEntry{},
+				},
+			},
+		}
+		err = r.Create(ctx, config)
+
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	contraintStatusList := &v1beta1.ConstraintPodStatusList{}
+	err = r.List(ctx, contraintStatusList, client.InNamespace(request.NamespacedName.Namespace))
+
+	if err != nil {
+		return reconcile.Result{Requeue: false}, err
+	}
+
+	constraintKindNames := map[string][]string{}
+
+	for _, c := range contraintStatusList.Items {
+		if !slices.Contains(c.Status.Operations, "audit") {
+			continue
+		}
+
+		labels := c.GetLabels()
+		//nolint:all
```

Comment:
I tried to cut but this was the best. hm.. I can assign in new variable 
```
kindKey := "internal.gatekeeper.sh/constraint-kind"
constraintKindNames[labels[kindKey]]
```

looks like useless memory assign so I tried to avoid this

Reply:
Ah, was it about duplicated strings? If so, I mean it seems better to do `//nolint:<name-of-linter>` rather than `//nolint:all`.

But I also personally think that new code you posted is easier to read, and the memory hit is minimal.

Reply:
It is just toooo long